### PR TITLE
caveats on temporary local addresses

### LINKED ISSUE #626: Problems with temporary local addresses
sec 5.2.12
Using temporary local addresses may break various mechanisms that prove address ownership (e.g QUIC resumption tokens) and therefore impair performance as the client has to re-verify its address.

### PR DESCRIPTION
**Pull request by britram**
fixes #626

### CODE REVIEW

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -1043,9 +1043,13 @@ This property allows the application to express a preference for the use of
...
 Temporary addresses are generally used to prevent linking connections over time
 when a stable address, sometimes called "permanent" address, is not needed.
-Note that if an application Requires the use of temporary addresses, the
-resulting Connection cannot use IPv4, as temporary addresses do not exist in
-IPv4.
+There are two caveats to note when specifying this property. First, if an
+application Requires the use of temporary addresses, the resulting Connection 
+cannot use IPv4, as temporary addresses do not exist in IPv4. Second, temporary
+local addresses may involve privacy/performance tradeoffs with protocol 
+stacks and networks that use session resumption in the presence of network
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md:1050]: I don't understand what is intended by "using a stack"... "in the presence of network address translation"

Comment (martinduke) [reply draft-ietf-taps-interface.md:1050]: I agree with Gorry. A NAT would have a similar impact as a temporary address; proof-of-address tokens would no longer function.

Comment (mwelzl) [reply draft-ietf-taps-interface.md:1050]: Comment from Martin: "in the presence of NAT" is wrong, should be removed

**Comment by britram** [comment]
@gorryfair PTAL

File: draft-ietf-taps-interface.md [hunk 2]
```diff
@@ -1048,9 +1048,12 @@ This property allows the application to express a preference for the use of
...
 when a stable address, sometimes called "permanent" address, is not needed.
-Note that if an application Requires the use of temporary addresses, the
-resulting Connection cannot use IPv4, because temporary addresses do not exist in
-IPv4.
+There are some caveats to note when specifying this property. First, if an
+application Requires the use of temporary addresses, the resulting Connection
+cannot use IPv4, as temporary addresses do not exist in IPv4. Second, temporary
+local addresses may involve trading off privacy for performance. For instance,
+temporary addresses can interfere with resumption mechanisms that some protocols
+rely on to reduce first-packet latency.
```
Comment (mirjak) [inline draft-ietf-taps-interface.md:1056]:
```suggestion
rely on to reduce initial latency.
```

File: draft-ietf-taps-interface.md [hunk 3]
```diff
@@ -1048,9 +1048,12 @@ This property allows the application to express a preference for the use of
 temporary local addresses, sometimes called "privacy" addresses {{!RFC4941}}.
 Temporary addresses are generally used to prevent linking connections over time
 when a stable address, sometimes called "permanent" address, is not needed.
-Note that if an application Requires the use of temporary addresses, the
-resulting Connection cannot use IPv4, because temporary addresses do not exist in
-IPv4.
+There are some caveats to note when specifying this property. First, if an
+application Requires the use of temporary addresses, the resulting Connection
+cannot use IPv4, as temporary addresses do not exist in IPv4. Second, temporary
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md:1053]: /as/because/ ... do not exist?

File: draft-ietf-taps-interface.md [hunk 4]
```diff
@@ -1048,9 +1048,12 @@ This property allows the application to express a preference for the use of
...
+local addresses may involve trading off privacy for performance. For instance,
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md:1054]: /may involve/might involve/

File: draft-ietf-taps-interface.md (hunk 2 above)
Comment (gorryfair) [inline draft-ietf-taps-interface.md:1056]: ... /first-packet/packet/ ....can it be more than the first packet?

**Review by gorryfair — COMMENTED** [review]
This looks pretty close ... check these words and submit.

**Pull request merged**